Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -124,8 +155,29 @@ async function doWork() {
             };
 
             var handler = new handlers.BasicCredentialHandler(username, password);
-            
-            await getArtifactsFromUrl(artifactQueryUrl, strictSSL, localPathRoot, itemPattern, handler, variables);
+            if (!itemPattern || itemPattern === '**') {
+                const archiveUrl: string = `${serverEndpointUrl}/${jenkinsJobDetails.jobUrlInfix}/${jenkinsJobDetails.multiBranchPipelineUrlInfix}/${jenkinsJobDetails.buildId}/artifact/*zip*/`
+                var zipLocation = path.join(localPathRoot, "archive.zip");
+                await getZipFromUrl(archiveUrl, strictSSL, zipLocation, handler);
+
+                var unzipPromise = unzip(zipLocation, localPathRoot);
+                unzipPromise.catch((error) => {
+                    throw error;
+                });
+                
+                await unzipPromise;
+
+                if (tl.exist(zipLocation)) {
+                    tl.rmRF(zipLocation);
+                }
+
+                fsExtra.move(path.join(localPathRoot, "archive"), localPathRoot).catch((error) => {
+                    console.error(error);
```

Comment:
why not throw error as you are doing above?

Reply:
fixed.